File: opensearch/src/test/java/io/micronaut/opensearch/OpenSearchClientFactoryTest.java
package io.micronaut.opensearch;

import io.micronaut.context.BeanContext;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;
import org.opensearch.client.opensearch.OpenSearchAsyncClient;
import org.opensearch.client.opensearch.OpenSearchClient;

import static org.junit.jupiter.api.Assertions.*;

@MicronautTest(startApplication = false)
class OpenSearchClientFactoryTest {
    @Inject
    BeanContext beanContext;

    @Test
    void beanOfTypeOpenSearchClientExists() {
        assertFalse(beanContext.containsBean(OpenSearchClient.class));
    }

    @Test
    void beanOfTypeOpenSearchAsyncClientExists() {
        assertFalse(beanContext.containsBean(OpenSearchAsyncClient.class));
    }
}

File: test-suite-python/src/test/python/micronaut/example/OpenSearchClientFactoryTest.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context import BeanProvider
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test
from org.opensearch.client.opensearch import OpenSearchAsyncClient, OpenSearchClient


@MicronautTest(startApplication=False)
class OpenSearchClientFactoryTest:
    client: Annotated[BeanProvider[OpenSearchClient], Inject]
    async_client: Annotated[BeanProvider[OpenSearchAsyncClient], Inject]

    @Test
    def test_bean_of_type_open_search_client_exists(self) -> None:
        assert self.client.isPresent()

    @Test
    def test_bean_of_type_open_search_async_client_exists(self) -> None:
        assert self.async_client.isPresent()
